docs/tutorials/particle-filtering.md

```diff
@@ -1,3 +1,564 @@
+
+```cpp
+const auto estimation = beluga::estimate(beluga::views::states(particles), beluga::views::weights(particles));
+```
+
+Here, `estimation` is a `std::tuple<double, double>` data structure.
+
+:::{attention}
+`estimation` is not used in this code. You can use `<iostream>` to print the value of `estimation` on the screen or complete the TODO: [Visualization tutorial]().
+:::
+
+## 4 Compile and run the code
+
+### 4.1 package.xml
+
+Create a `beluga_tutorial/package.xml` file and put the following inside:
+
+```xml
+<?xml version="1.0"?>
+<?xml-model href="http://download.ros.org/schema/package_format3.xsd" schematypens="http://www.w3.org/2001/XMLSchema"?>
+<package format="3">
+  <name>beluga_tutorial</name>
+  <version>0.0.0</version>
+  <description>TODO: Package description</description>
+  <maintainer email="user@mail.com">developer</maintainer>
+  <license>Apache License 2.0</license>
+
+  <buildtool_depend>cmake</buildtool_depend>
+
+  <depend>beluga</depend>
+
+  <export>
+    <build_type>cmake</build_type>
+  </export>
+</package>
+```
+
+This declares the package needs `beluga` when its code is built and executed.
+
+### 4.2 CMakeLists.txt
+
+The final `beluga_tutorial/CMakeLists.txt` looks like:
+
+```cmake
+cmake_minimum_required(VERSION 3.16)
+
+project(beluga_tutorial)
+
+find_package(beluga REQUIRED)
+
+add_executable(${PROJECT_NAME} src/main.cpp)
+
+target_link_libraries(${PROJECT_NAME} beluga::beluga)
+
+install(TARGETS ${PROJECT_NAME} RUNTIME DESTINATION lib/${PROJECT_NAME})
+```
+
+* `project()` define the name of the project. `${PROJECT_NAME}` macro will use this name alongside the file.
+* `find_package()` is used to import external dependencies.
+* `add_executable()` create the executable called `beluga_tutorial`.
+* `target_link_libraries()` link `beluga` in the compilation process.
+* `install(TARGETS...)` will install the executable under the path `install/beluga_tutorial/lib/beluga_tutorial/`.
+
+### 4.2 Build and run
+
+Open a new terminal and build your new package:
+
+```bash
+source /opt/ros/humble/setup.bash
+cd beluga_ws/
+colcon build
+```
+
+In the same terminal run the code:
+
+```bash
+./install/beluga_tutorial/lib/beluga_tutorial/beluga_tutorial
+```
+
+## Conclusion
+
+You can see that we have put most of our effort into explaining and implementing the components that make up or are used by the particle filter, such as the landmark map, motion model, sensor model, and how to initialize the set of particles. This was made possible thanks to **Beluga**, a C++ library that provides a set of reusable components that can be combined to assemble a complete and customizable particle filter. Moreover, **Beluga** offers additional features that can help build more complex {abbr}`MCL (Monte Carlo Localiaztion)` algorithms than those proposed by this tutorial, while minimizing code duplication and maximizing code reuse.
```

Comment:
This is part of the Beluga documentation. It doesn't seem necessary to explain what Beluga is.